qds_sdk/cluster.py

```diff
@@ -1168,6 +1214,217 @@ def minimal_payload(self):
+            except Exception as e:
+                raise Exception("Invalid JSON string for custom ec2 tags: %s" % e.message)
+        self.set_data_disk(size, count, type, upscaling_config, encryption)
+        self.cluster_info['heterogeneous_config'] = heterogeneous_config
+        self.cluster_info['slave_request_type'] = slave_request_type
+        self.set_spot_settings(maximum_bid_price_percentage, timeout_for_request,
+                                        maximum_spot_instance_percentage, stable_maximum_bid_price_percentage,
+                                        stable_timeout_for_request, stable_allow_fallback)
+
+    def set_spot_settings(self, maximum_bid_price_percentage=None, timeout_for_request=None,
+                          maximum_spot_instance_percentage=None, stable_maximum_bid_price_percentage=None,
+                          stable_timeout_for_request=None, stable_allow_fallback=None):
+        self.cluster_info['spot_settings'] = {}
+        self.set_spot_instance_settings(maximum_bid_price_percentage, timeout_for_request,
+                                        maximum_spot_instance_percentage)
+        self.set_stable_spot_bid_settings(stable_maximum_bid_price_percentage, stable_timeout_for_request,
+                                          stable_allow_fallback)
+
+    def set_spot_instance_settings(self, maximum_bid_price_percentage=None, timeout_for_request=None,
+                          maximum_spot_instance_percentage=None):
+        self.cluster_info['spot_settings']['spot_instance_settings'] = {}
+        self.cluster_info['spot_settings']['spot_instance_settings']['maximum_bid_price_percentage'] = maximum_bid_price_percentage
+        self.cluster_info['spot_settings']['spot_instance_settings']['timeout_for_request'] = timeout_for_request
+        self.cluster_info['spot_settings']['spot_instance_settings']['maximum_spot_instance_percentage'] = maximum_spot_instance_percentage
+
+    def set_stable_spot_bid_settings(self, stable_maximum_bid_price_percentage=None,
+                          stable_timeout_for_request=None, stable_allow_fallback=None):
+        self.cluster_info['spot_settings']['stable_spot_bid_settings'] = {}
+        self.cluster_info['spot_settings']['stable_spot_bid_settings']['stable_maximum_bid_price_percentage'] = stable_maximum_bid_price_percentage
+        self.cluster_info['spot_settings']['stable_spot_bid_settings']['stable_timeout_for_request'] = stable_timeout_for_request
+        self.cluster_info['spot_settings']['stable_spot_bid_settings']['stable_allow_fallback'] = stable_allow_fallback
+
+    def set_data_disk(self, size=None, count=None, type=None, upscaling_config=None, encryption=None):
+        self.cluster_info['datadisk'] = {}
+        self.cluster_info['datadisk']['size'] = size
+        self.cluster_info['datadisk']['count'] = count
+        self.cluster_info['datadisk']['type'] = type
```

Comment:
Don't use `type` as a field name. That shadows the built-in Python `type` and can undesirable effects.